runtime/src/commonMain/kotlin/pbandk/internal/json/JsonValueUnmarshaller.kt

```diff
@@ -46,29 +47,58 @@ internal class JsonValueUnmarshaller(private val jsonConfig: JsonConfig) {
         is FieldDescriptor.Type.Map<*, *> -> readMap(value, type)
     }
 
-
-    fun readNumber32(value: JsonElement): Int = try {
-        value.int
+    private inline fun <T> readNumberInternal(
```

Comment:
🔧 You may want to change this to something like `readIntegerInternal`. When I read this method at first, ` '-' -> if (content.getOrNull(1) == '0')` jumped out to me as a bug for numbers like `-0.1` After looking into the code more and seeing this was only used for integers I saw otherwise, but a name change could help out earlier.

Reply:
Good idea. Will do!